### PR TITLE
Initial component setup vr

### PR DESCRIPTION
**Pull request by vrandall66**
## What does this PR do?
This PR adds six apiCalls functions (four GETS, two POSTS), installs react-icons and react-modal npm packages, implements React Router and builds out the HomeNav, Nav, Project Nav and the PaletteForm components.
### Where should the reviewer start?
Start in apiCalls to see the methods I wrote and began to import into the App.js file. After that, I would start at the various Nav components to see the initial structure I have given them. Right now they are not successfully rendering, but fresh eyes in the morning should quickly sort that out.
I have made minor changes in all Component files to give them consistent BEM class names for styling when we get there. I have added very minor flex box styling additions to the homepage, but want to discuss how we plan to navigate styling the projects container page. I can explain more during stand up.
#### How should this be manually tested?
Pull down and view the minor styling adjustments!
#### Any background context you want to provide?
I have only imported the `getAllProjects` method from apiCalls.js because when I was trying to set the results to state, I was having issues with rendering the information that does not exist. I think this might relate to the `plan to navigate styling the projects container page` comment I made above. Again, fresh eyes will probably quickly resolve this issue. We are still relying on the hardcoded data in state to render information.
#### What are the relevant tickets?
#19 , #20 , #21 
#### Screenshots (if appropriate)
Screen Shot 2019-12-08 at 11 06 17 PM

#### Additional notes

I have tried to rebase to master already, so try rebasing this PR and let me know the verdict.

### CODE REVIEW

File: src/Components/HomeNav/HomeNav.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import { IoMdRefresh } from 'react-icons/io';
+import { GiSave } from 'react-icons/gi';
+import { FaRegFolder } from 'react-icons/fa';
+import './HomeNav.scss';
+
+export class HomeNav extends Component {
+  render() {
+    return (
+      <div className='HomeNav'>
+        <IoMdRefresh />
```
Comment (vrandall66) [inline src/Components/HomeNav/HomeNav.js]: Once we figure out the rendering issue, we should have some large icons on our Nav!

File: src/Components/ColorCard/ColorCard.js [hunk 2]
```diff
@@ -1,21 +1,28 @@
...
   render() {
     return (
-    <div>
-      <h4 style={{backgroundColor: this.state.hex}}>{ this.state.hex }</h4>
-    </div>
-    )
+      <div className='ColorCard' style={{ backgroundColor: this.state.hex }}>
+        <h4 className='ColorCard__h4'>{this.state.hex}</h4>
+        {this.state.selected ? (
+          <IoMdCheckboxOutline />
```
Comment (vrandall66) [inline src/Components/ColorCard/ColorCard.js]: I've added icons for our eventual `selected` property in state!

File: src/Components/App/App.js [hunk 3]
```diff
@@ -1,77 +1,111 @@
...
-        {/* <Nav /> */}
-        {/* { projects } */}
-        <Container hexCodes = {this.state.hexCodes}/>
+        <Switch>
+          <Route
+            exact
+            path='/'
+            render={() => (
+              <>
+                <Nav>
```
Comment (vrandall66) [inline src/Components/App/App.js]: For some reason, the Nav is not currently rendering the HomeNav or ProjectNav components.

File: src/Components/ColorCard/ColorCard.js (hunk 2 above)
Comment (jmevans0211) [reply src/Components/ColorCard/ColorCard.js]: Excellent!

File: src/Components/Container/Container.js [hunk 4]
```diff
@@ -1,18 +1,17 @@
 import React from 'react';
 import ColorCard from './../ColorCard/ColorCard.js';
+import './Container.scss';
 
 const Container = ({ hexCodes }) => {
-  const colorCards = hexCodes.map(hex => {
-    return <ColorCard hex={hex}/>
-  })
+  const colorCards = hexCodes.map((hex, index) => {
+    return <ColorCard key={index} hex={hex} />;
```
Comment (jmevans0211) [inline src/Components/Container/Container.js]: Thanks for this addition.

File: src/Components/HomeNav/HomeNav.js (hunk 1 above)
Comment (jmevans0211) [reply src/Components/HomeNav/HomeNav.js]: I am ok with our decision to have two Navs, which is actually better than three! They do different things entirely, so I am ok with it. In the iteration to add users, we will be able to conditionally render buttons/icons more easily.

**Pull request merged**